atlasdb-commons/src/main/java/com/palantir/common/concurrent/ConcurrentStreams.java

```diff
@@ -113,6 +113,24 @@ private ConcurrentStreams() {}
         public synchronized Throwable fillInStackTrace() {
             return this;
         }
-    };
+    }
+
+    private static class StreamElement<T, S> {
```

Comment:
Should we make this immutable?

Reply:
Fine with me. Is it okay to pull that dependency into `atlasdb-commons`?

Reply:
I think it should be okay.